src/CellData/SkeletonCellFieldPair.jl

```diff
@@ -80,12 +76,28 @@ is passed into it. Ideally, if we could parse and extract only the Skeleton
 integration terms from the functional's julia function form, this fix is not
 required, but this is not trivial to do. On the positive side, since the
 evaluations are all lazy and not used, this doesn't put any noticable memory
-or computational overhead.
+or computational overhead. Ofcourse, it is made sure that the such plus side
+pick doesn't happen when the integration over the SkeletonTriangulation
 =#
-function change_domain(a::SkeletonCellFieldPair,target_trian::Triangulation,target_domain::DomainStyle)
-  change_domain(a.plus,DomainStyle(a),target_trian,target_domain)
+# If SkeletonCellFieldPair is evaluated we just pick the plus side parent
+function evaluate!(cache,f::SkeletonCellFieldPair,x::CellPoint)
+  _f, _x = _to_common_domain(f.plus.parent,x)
+  cell_field = get_data(_f)
+  cell_point = get_data(_x)
+  lazy_map(evaluate,cell_field,cell_point)
+end
+
+# fix for CellFieldAt{T}(parent::OperationCellField) giving right output
+# or else it chooses the SkeletonCellFieldPair directly as the parent
+# resulting in errors and it not the intended behaviour to have the parent
+# as SkeletonCellFieldPair and is not consistent with our getproperty rules
+function CellFieldAt{T}(parent::SkeletonCellFieldPair) where T
+  getproperty(parent,T)
 end
 
+# to handle the evaluation of SkeletonCellFieldPair at BoundaryTriangulation
```

Comment:
Ok. I do not see any problem in overriding `get_data`. Indeed, this is a method which belong to the abstract interface of a `CellField`. My only concern is ... are we 100% sure that this behaviour of `get_data` is the intended one in any of the calls to `get_data(::SkeletonCellFieldPair)`? (To think).

Reply:
Yes true, I am was unsure about this fully. I checked for the cases of all trians if there could be a side effect, but there was wasn't.

Reply:
The `get_data` is called in `change_domain_ref_ref` and `change_domain_phy_phs` when the `trian` for integrand and `quad` are not the same. This the case for `BoundaryTriangulation` and `SkeletonTriangulation`, choosing `plus` side is consistent with that of the choice we make in `evaluation!`. In case of integration over `SkeletonTriangulation` which hits inside `change_domain_ref_ref` specialised for `SkeletonTriangulation` the call `isa(get_array(a.cell_field),Fill)` gives error for `uh` and also `SkeletonCellFieldPair` so `get_data` doesn't effect here. In case of `BodyFittedTriangulation` there is no hit to `get_data` directly and `evaluate!` handles it as intended

Reply:
ok, so we can go ahead. Thanks for looking deeply into this.